web/src/engine/websites/Anchira.ts

```diff
@@ -0,0 +1,95 @@
+import { Tags } from '../Tags';
+import icon from './Anchira.webp';
+import { type Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+import { Priority, TaskPool } from '../taskpool/TaskPool';
+import { RateLimit } from '../taskpool/RateLimit';
+
+type APIMangas = {
+    entries: APIManga[]
+}
+
+type APIManga = {
+    id: number,
+    key: string,
+    title: string
+    hash?: string,
+    names?: string[]
+}
+
+@Common.ChaptersUniqueFromManga()
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly mangasTaskPool = new TaskPool(4, new RateLimit(8, 1));
+    private readonly apiUrl = new URL('/api/v1/', this.URI).href;
+    private readonly imageCDN = 'https://kisakisexo.xyz';
+    private readonly mangaRegexp = new RegExp(`^${this.URI.origin}/g/([\\d]+/[^/]+)$`);
+
+    public constructor() {
+        super('anchira', `Anchira`, 'https://anchira.to', Tags.Language.English, Tags.Media.Manga, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return this.mangaRegexp.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const matches = url.match(this.mangaRegexp);
+        const item = await FetchJSON<APIManga>(new Request(new URL(`${this.apiUrl}library/${matches[1]}`), {
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest',
+                'Referer': `${this.URI}?`
+            }
+        }));
+        return new Manga(this, provider, `${item.id}/${item.key}`, item.title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const cancellator = new AbortController();
+        try {
+            const mangaList = [];
+            for (let page = 1, run = true; run; page++) {
+                const mangas = await this.GetMangasFromPage(page, provider, cancellator.signal);
```

Comment:
Using a `taskpool` is redundant when only adding a new task after the previous task is completed.
The intended use is to push all tasks at once into the `taskpool` and then await all the promises to be fulfilled.

See: https://github.com/manga-download/haruneko/blob/master/web/src/engine/websites/Piccoma.ts#L67-L71

Reply:
I get it now, sorry